src/NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/BoundaryFlux.hpp

```diff
@@ -0,0 +1,92 @@
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/BoundaryData.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/LiftFlux.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/NumericalFluxes/NumericalFluxHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Protocols.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace dg {
+/// Functionality related to the strong first-order boundary scheme.
+namespace StrongFirstOrder {
+
+/*!
+ * \brief Compute the boundary flux contribution and lift it to the volume.
+ *
+ * This computes the numerical flux minus the flux dotted into the interface
+ * normal, projects the result to the face mesh if necessary, and then lifts it
+ * to the volume (still presented only on the face mesh as all other points are
+ * zero).
+ *
+ * Projection must happen after the numerical flux calculation so that the
+ * elements on either side of the mortar calculate the same result.  Projection
+ * must happen before flux lifting because we want the factor of the magnitude
+ * of the unit normal added during the lift to cancel the Jacobian factor in
+ * integrals to preserve conservation; this only happens if the two operations
+ * are done on the same grid.
+ */
+template <size_t FaceDim, typename NumericalFluxType,
+          Requires<tt::conforms_to_v<NumericalFluxType,
+                                     dg::protocols::NumericalFlux>> = nullptr>
+Variables<typename NumericalFluxType::variables_tags> boundary_flux(
+    const BoundaryData<NumericalFluxType>& local_boundary_data,
+    const BoundaryData<NumericalFluxType>& remote_boundary_data,
+    const NumericalFluxType& numerical_flux_computer,
+    const Scalar<DataVector>& magnitude_of_face_normal,
+    const size_t extent_perpendicular_to_boundary,
+    const Mesh<FaceDim>& face_mesh, const Mesh<FaceDim>& mortar_mesh,
+    const MortarSize<FaceDim>& mortar_size) noexcept {
+  using variables_tags = typename NumericalFluxType::variables_tags;
+
+  // Compute the Tags::NormalDotNumericalFlux from local and remote data
+  Variables<db::wrap_tags_in<::Tags::NormalDotNumericalFlux, variables_tags>>
+      normal_dot_numerical_fluxes{mortar_mesh.number_of_grid_points()};
+  dg::NumericalFluxes::normal_dot_numerical_fluxes(
+      make_not_null(&normal_dot_numerical_fluxes), numerical_flux_computer,
+      local_boundary_data, remote_boundary_data);
+
+  // Subtract the local Tags::NormalDotFlux
+  tmpl::for_each<variables_tags>([&normal_dot_numerical_fluxes,
```

Comment:
In terms of dealing with the weak form, I was going to wrap this `tmpl::for_each` in an `if` statement and take the `dg::Formulation` as an explicit template parameter, so that won't be an issue.